### DISCUSSION TITLE
How to specify "unnamed" uint arg, e.g. `git log -10`

### DISCUSSION THREAD

**Discussion by vallentin** [pacak/bpaf#347] [Q&A] [answered]
I've been looking around the API and the examples, trying to figure out if it's possible to replicate an arg like `-<uint>`.

Similarly to `git log -<uint>`, e.g. `git log -10` for [commit limiting](https://git-scm.com/docs/git-log#_commit_limiting). Just to clarify, this is not a positional arg.

Preferably using the derive API, otherwise using the combinatoric API.

**Comment by pacak** [comment, marked as answer]
[`any`](https://docs.rs/bpaf/0.9.9/bpaf/fn.any.html) + [`ParseAny::anywhere`](https://docs.rs/bpaf/0.9.9/bpaf/parsers/struct.ParseAny.html#method.anywhere) should do the trick. You can find a complete example [here](https://github.com/pacak/bpaf/blob/4141714cd165dddd5146182bc67097a339b4bc8b/examples/compression.rs)

**Reply by pacak** [reply]
You can use them in both Combinatoric and Derive APIs, but personally I'd write a standalone function `fn depth() -> impl Parser<usize>` and use it from derive API with `external`

**Reply by vallentin** [reply]
Thanks for the literally instant response! I don't even know how I overlooked that...

I can get it to work using combinatoric:

```rust
#[derive(Bpaf, Clone, Debug)]
#[bpaf(options)]
pub struct Args {
    #[bpaf(external(limit))]
    n: usize,
    foo: bool,
}

fn limit() -> impl Parser<usize> {
    bpaf::any::<isize, _, _>("LIMIT", isize_to_usize)
        .help("LIMIT")
        .anywhere()
}

fn isize_to_usize(n: isize) -> Option<usize> {
    n.abs().try_into().ok()
}
```

However, I'm confused how to specify `anywhere()` using derive? (Such that it can be _anywhere_ and isn't a right-most positional)

```rust
#[derive(Bpaf, Clone, Debug)]
#[bpaf(options)]
pub struct Args {
    // anywhere?
    #[bpaf(any::<isize>("LIMIT", isize_to_usize))]
    n: usize,
    foo: bool,
}

fn isize_to_usize(n: isize) -> Option<usize> {
    n.abs().try_into().ok()
}
```

**Reply by pacak** [reply]
Should be this I think: `#[bpaf(any::<isize>("LIMIT", isize_to_usize), anywhere)]`

It is easier to think about items in `#[bpaf(XXX)]` annotations as chained calls combined with `.`.

For example this: `#[bpaf(external(backing), fallback(false), debug_fallback)]` is really this: `backing().fallback(false).debug_fallback()`.

**Reply by vallentin** [reply]
That sadly gives a compile error:

```console
error: Unexpected attribute in field annotation
  --> src\main.rs:14:51
   |
14 |     #[bpaf(any::<isize>("LIMIT", isize_to_usize), anywhere)]
   |                                                   ^^^^^^^^
```

**Reply by pacak** [reply]
(came back from a run)

My bad, turns out `anywhere` is not supported at all at the moment by derive macro. Will make a fix shortly.

**Reply by vallentin** [reply]
I will use the combinatoric approach for now then. Thanks! :)
